### DISCUSSION TITLE
Ability to control automerge not by the age of a dependency (`minimumReleaseAge`) but by the age of a PR

### DISCUSSION THREAD

**Discussion by maxdanilov** [renovatebot/renovate#23743] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have the following use case for `packageRules` that I can't make work:
- `schedule` is configured to create branches every `2 weeks`
- `automerge` is enabled
- between a PR is created and the actual automerging, at least N hours are passed
- PRs are not left open for a long period of time before being automerged

I want to be able to control the auto-merge, so it does not happen immediately after the PR creation (Renovate needs to wait for all checks to complete. Some of my checks take a while to get registered at GitHub and Renovate merges too fast). 

Currently, the only similar config parameter for this I've found is [minimumReleaseAge](https://docs.renovatebot.com/configuration-options/#minimumreleaseage), however, I don't think it covers my use case:
- `releaseAge` is the age of the package (e.g. how long ago it has been released on GitHub/npm etc.), not the age of an actual PR
- in case of `automerge` enabled (see this [note](https://docs.renovatebot.com/configuration-options/#await-x-time-duration-before-automerging)):
> If you enabled automerge and minimumReleaseAge, it means that PRs will be created immediately but automerging will be delayed until the time-duration has passed. This works because Renovate will add a "renovate/stability-days" pending status check to each branch/PR and that pending check will prevent the branch going green to automerge.

which in practice means that, for example, if a dependency was released 10 days ago, and Renovate creates a PR now (after 2 weeks from the previous check), the `releaseAge` of the dependency is counted as 10 days and unless I put `minimumReleaseAge` to something higher than 10 days, it will be automerged before waiting for the CI checks to run (as soon as a PR is created).
- putting `minimumReleaseAge` to a high value though means that PRs have to likely stay open for very long for no good reason
  - for example, if a dependency was released 1 day ago and Renovate creates a PR now and `minimumReleaseAge` is set to 10 days, the PR won't be automerged for another 9 days

### What you tried so far.

`minimumReleaseAge` in combination with `automerge`

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
If you're using the hosted app then it's probably platformAutomerge (i.e. GitHub automerge) and not Renovate directly?

**Comment by viceice** [comment]
make those checks mandatory on GitHub

**Reply by maxdanilov** [reply]
the checks are different based on what are the code changes (e.g. Terraform Cloud checks only run if Terraform code is changed in a PR), unfortunately, I can't make all of them mandatory.

**Comment by maxdanilov** [comment]
> If you're using the hosted app then it's probably platformAutomerge (i.e. GitHub automerge) and not Renovate directly?

`platformAutomerge` is set to true (default value), but I'm not sure how changing it to `false` will help Renovate to delay a PR automerge in the same run where a PR is created.

**Reply by rarkins** [reply, marked as answer]
When platformAutomerge=false then it usual takes a minimum few hours to automerge
